tcms/testplans/tests/test_new_plan.py

```diff
@@ -1,44 +1,57 @@
 from http import HTTPStatus
 
 from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from uuslug import slugify
 
 from tcms.testplans.models import TestPlan
-from tcms.testplans.tests.tests import BasePlanTest
-from tcms.tests import remove_perm_from_user, user_should_have_perm
-
-
-class NewPlanViewTest(BasePlanTest):
-    location = reverse('plans-new')
-    plan_name = 'plan name'
-    add_testplan_permission = 'testplans.add_testplan'
-
-    def setUp(self):
-        super().setUp()
-
-        user_should_have_perm(self.user, perm=self.add_testplan_permission)
-
-        self.user.is_superuser = False
-        self.user.save()
-
-        self.request = {
-            'author': self.user.pk,
-            'product': self.product.pk,
-            'product_version': self.product_version.pk,
-            'type': self.plan_type.pk,
-            'name': self.plan_name,
-
-            'email_settings-0-auto_to_plan_author': 'on',
-            'email_settings-0-auto_to_case_owner': 'on',
-            'email_settings-0-auto_to_case_default_tester': 'on',
-            'email_settings-0-notify_on_case_update': 'on',
-            'email_settings-0-notify_on_plan_update': 'on',
-
-            'email_settings-0-plan': '',
-            'email_settings-0-id': self.test_plan.emailing.pk,
-            'email_settings-TOTAL_FORMS': '1',
-            'email_settings-INITIAL_FORMS': '1',
-            'email_settings-MIN_NUM_FORMS': '0',
-            'email_settings-MAX_NUM_FORMS': '1',
+from tcms.tests import (
+    user_should_have_perm,
+    LoggedInTestCase,
+    PermissionsTestCase,
+)
+from tcms.tests.factories import (
+    TestPlanFactory,
+    ProductFactory,
+    VersionFactory,
+    PlanTypeFactory,
+)
+
+
+class TestNewTestPlanView(LoggedInTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.plan_name = 'plan_name'
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.plan_type = PlanTypeFactory()
+        cls.test_plan = TestPlanFactory()
+
+        user_should_have_perm(cls.tester, perm="testplans.add_testplan")
+        user_should_have_perm(cls.tester, perm="testplans.view_testplan")
+
+        cls.location = reverse("plans-new")
+
+        cls.request = {
+            "author": cls.tester.pk,
+            "product": cls.product.pk,
+            "product_version": cls.product_version.pk,
+            "type": cls.plan_type.pk,
+            "name": cls.plan_name,
```

Comment:
The rest of the code in this class isn't using the class attribute so this can be hard-coded value. Also can use the class name as the value (or in combination with string format) to give it more uniqueness because in the test class for permissions you are using the same name and then searching by name.